modular_skyrat/modules/ash_farming/code/ash_farming.dm

```diff
@@ -29,6 +33,36 @@
 		to_chat(user, span_notice("You finish planting."))
 		qdel(attacking_grown)
 		return
+	else if(istype(attacking_object, /obj/item/grown))
+		var/planting_chance = 30 //you only have a 30 percent chance of succeeding in planting, unless you are an ashwalker
+		if(iscarbon(user))
+			var/mob/living/carbon/carbon_user = user
+			if(carbon_user?.dna?.species?.id == SPECIES_LIZARD_ASH)
+				planting_chance = 80
+		var/obj/item/grown/attacking_grown = attacking_object
+		if(!dug)
+			to_chat(user, span_warning("You cannot grow plants on [src] without it being dug first!"))
+			return
+		if(!attacking_grown.seed)
+			to_chat(user, span_warning("[attacking_grown] does not have a seed, it cannot be grown!"))
+			return
+		to_chat(user, span_notice("You begin planting..."))
+		if(!do_after(user, 5 SECONDS, target = src))
+			to_chat(user, span_warning("You interrupt your planting!"))
+			return
+		if(!prob(planting_chance))
+			to_chat(user, span_warning("[attacking_grown] breaks in your hands!"))
+			qdel(attacking_grown)
+			return
+		var/obj/structure/flora/ash_farming/new_farm = new /obj/structure/flora/ash_farming(src)
+		new_farm.planted_seeds = attacking_grown.seed
+		new_farm.name = new_farm.planted_seeds.plantname
+		new_farm.icon = new_farm.planted_seeds.growing_icon
+		new_farm.icon_state = "[new_farm.planted_seeds.icon_grow]1"
+		new_farm.update_appearance()
+		to_chat(user, span_notice("You finish planting."))
+		qdel(attacking_grown)
+		return
```

Comment:
Surely you can find a way to incorporate the check for whether it's an `item/grown` or an `item/food/grown` without having to duplicate all of this code.

Reply:
rude!